src/v/security/gssapi_principal_mapper.h

```diff
@@ -108,6 +109,26 @@ class gssapi_rule {
     make_lower _to_lower{false};
     make_upper _to_upper{false};
 };
+
+class gssapi_principal_mapper {
+public:
+    gssapi_principal_mapper(
+      std::string_view default_realm,
+      config::binding<std::optional<std::vector<ss::sstring>>>
+        principal_to_local_rules_cb);
+    std::optional<ss::sstring> apply(gssapi_name name) const;
+
+private:
+    friend struct fmt::formatter<gssapi_principal_mapper>;
+
+    friend std::ostream&
+    operator<<(std::ostream& os, const gssapi_principal_mapper& p);
```

Comment:
i'm curious: if a fmt::formatter specialization is defined, is the output operator needed?

Reply:
No it is not.  Just commented it out and built/ran unit tests and they ran fine.  I'll remove them.

Reply:
Actually nix that.  I removed the one for `gssapi_rule` and it didn't build.  I need to look a little bit more into how fmt works with this.  My only experience with fmt is using it with `spdlog`

Reply:
i wouldn't waste much time on it if it is working. at this point it's a curiosity if we can have one or the other in the future.